unified-installer/astra-unified-installer.sh

```diff
@@ -99,15 +99,38 @@ readonly __FATAL=50
 #     captured_stdout="$(curl -sS https://bad-url.com 2> "$_ERR_FILE")"
 #     captured_stderr="$(get_captured_err)"
 readonly __ERR_FILE="tmp_last_captured_error.txt"
+
+# __TMP_ENV is used to store the user's env vars so that we can then re-apply them after having sourced
+# their config file. This allows us to make command line vars take precedence over what's in the config file.
+readonly __TMP_ENV="tmp.env"
+
 readonly __NEWLINE=$'\n' # This is for readability
 
 #----------------------------------------------------------------------
 #-- Script config
 #----------------------------------------------------------------------
+process_args() {
+    for arg in "$@"; do
+        case $arg in
+            --help|-h)
+                print_help
+                exit 0
+                ;;
+            --required)
+                local msg="Listed below are all required environment variables which do not have a default value"
+                msg+=" (and so must be provided by the user):${__NEWLINE}"
+                echo "$msg"
+                print_help | grep '\*[A-Z]\+'
+                exit 0
+                ;;
+        esac
+    done
+}
+
 get_configs() {
     # ------------ SCRIPT BEHAVIOR ------------
     CONFIG_FILE="${CONFIG_FILE:-}" # Overrides environment variables specified via command line
-    DRY_RUN="${DRY_RUN:-"true"}" # Skips applying generated resources
+    DRY_RUN="${DRY_RUN:-"false"}" # Skips applying generated resources
```

Comment:
PM request to default this to false, though it's been mentioned a few times by a few of you as well.